### PR TITLE
Feature/DGI-71: Search bar to search clients

### PR DESCRIPTION
**Pull request by ntw-ITM**
Very primitive search functionality. Could definitely need improvement with typeahead etc.

### CODE REVIEW

File: frontend/src/components/Accounts/Clients.tsx [hunk 1]
```diff
@@ -13,11 +13,13 @@ import { logger } from "utils/logger";
 import AccountList from "./AccountList/AccountList";
 import DownloadCsvBtn from "./DownloadCsvBtn";
 import NewAccountModal from "./NewAccountModal";
+import SearchBar from "./SearchBar";
 
 const Accounts: FC = () => {
   const { t } = useLocales();
 
   const [clients, dispatchClients] = useReducer(ListReducer<IClientDto>("id"), []);
+  const [filteredClients, setFilteredClients] = useState<IClientDto[]>(clients);
```
Comment (objmj-itminds) [inline frontend/src/components/Accounts/Clients.tsx:22]: Like a filter, it is best to save the predicate and not a copy of the entire data set.

File: frontend/src/components/Accounts/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }, [filteredClients]);
+
+  return (
+    <InputGroup w="15em">
+      <InputLeftElement>
+        <BiSearch opacity={0.5} />
+      </InputLeftElement>
+      <Input
+        value={searchString}
+        onChange={e => setSearchString(e.target.value)}
```
Comment (objmj-itminds) [inline frontend/src/components/Accounts/SearchBar.tsx:43]: A small delay of 200 ms and a limit of a minimum of 3 characters would help the search handle big amounts of data.

**Review by objmj-itminds — CHANGES_REQUESTED** [review]

File: frontend/src/components/Accounts/SearchBar.tsx (hunk 2 above)
Comment (ntw-ITM) [reply frontend/src/components/Accounts/SearchBar.tsx]: I have now added a 200 ms delay and minimum of 3 characters.

File: frontend/src/components/Accounts/Clients.tsx (hunk 1 above)
Comment (ntw-ITM) [reply frontend/src/components/Accounts/Clients.tsx]: I have now added an array of filters to Clients.tsx. For now it only contains one filter that uses the search string.

**Pull request merged**